addons/hr_expense/tests/test_expenses_states.py

```diff
@@ -0,0 +1,313 @@
+            {'payment_mode': 'own_account', 'state': 'approved'},
+            {'payment_mode': 'company_account', 'state': 'done'},
+        ])
+        self.assertRecordValues(self.expense_states_sheets, [
+            {'payment_mode': 'own_account', 'state': 'post', 'payment_state': 'not_paid'},
+            {'payment_mode': 'company_account', 'state': 'done', 'payment_state': 'paid'},
+        ])
+        self.assertRecordValues(self.expense_states_sheets.account_move_id, [
+            {'state': 'posted'},
+            {'state': 'posted'},
+        ])
+
+    def test_expense_state_synchro_2_employee_specific_flow(self):
+        self.expense_states_sheets.action_submit_sheet()
+        self.expense_states_sheets.approve_expense_sheets()
+        self.expense_states_sheets.action_sheet_move_create()
+
+        # STEP 1: ER posted -> Reset move to draft
+        self.expense_states_employee_sheet.account_move_id.button_draft()
+        self.assertRecordValues(self.expense_states_employee_sheet.expense_line_ids, [
+            {'state': 'approved'},
+        ])
+        self.assertRecordValues(self.expense_states_employee_sheet, [
+            {'state': 'post', 'payment_state': 'not_paid'},
+        ])
+        self.assertRecordValues(self.expense_states_employee_sheet.account_move_id, [
+            {'state': 'draft'},
+        ])
+
+        # STEP 2: ER posted with draft move -> Cancel move (nothing changes)
+        self.expense_states_employee_sheet.account_move_id.button_cancel()
+        self.assertRecordValues(self.expense_states_employee_sheet.expense_line_ids, [
+            {'state': 'approved'},
+        ])
+        self.assertRecordValues(self.expense_states_employee_sheet, [
+            {'state': 'post', 'payment_state': 'not_paid'},
+        ])
+        self.assertRecordValues(self.expense_states_employee_sheet.account_move_id, [
+            {'state': 'cancel'},
+        ])
+
+        # Change move state to draft
+        self.expense_states_employee_sheet.account_move_id.button_draft()
+
+        # STEP 3: ER posted with draft move -> unlink move (Reverts to approve state)
+        self.expense_states_employee_sheet.account_move_id.unlink()
+        self.assertRecordValues(self.expense_states_employee_sheet.expense_line_ids, [
+            {'state': 'approved'},
+        ])
+        self.assertRecordValues(self.expense_states_employee_sheet, [
+            {'state': 'approve', 'payment_state': 'not_paid'},
+        ])
+        self.assertFalse(self.expense_states_employee_sheet.account_move_id)
+
+        # Re-create posted move
+        self.expense_states_employee_sheet.action_sheet_move_create()
+
+        # STEP 4: ER with draft move -> Reverse move (Reverts to approve state)
+        self.expense_states_employee_sheet.account_move_id._reverse_moves(
+            default_values_list=[{'invoice_date': fields.Date.today(self.expense_states_employee_sheet)}],
```

Comment:
Why are we passing the sheet as argument for `today()` function? 
It useless with `today()` but not with `context_today()`.

Reply:
Because I'm an idiot I guess, that's what I wanted to do :facepalm: